include/net/nrf_cloud_codec.h

```diff
@@ -34,6 +34,7 @@ enum nrf_cloud_obj_type {
 	NRF_CLOUD_OBJ_TYPE__UNDEFINED,
 
 	NRF_CLOUD_OBJ_TYPE_JSON,
+	NRF_CLOUD_OBJ_TYPE_COAP_CBOR,
```

Comment:
it might be useful to explicitly state that this object type is to be used for only one type (nrf_cloud_data_type) of nrf cloud coap message... bulk and custom/nested messages are not supported.
not sure if the comment should be added here, on `struct nrf_cloud_obj_coap_cbor` or on `NRF_CLOUD_OBJ_COAP_CBOR_DEFINE`

Reply:
In addition to explicitly stating it with a comment as you suggest, we could also name the type itself something like `NRF_CLOUD_OBJ_TYPE_CBOR_DATA_MSG`. At minimum, it doesn't seem necessary to include `COAP` in the title, at least

Reply:
this type is only for nRF Cloud **_CoAP_** CBOR messages...

Reply:
i was explaining why i named it with the added COAP text and feel that it should stay.
we could potentially add a generic CBOR type later (which was kind of the original intent).

i don't think the JSON type is MQTT specific.